src/core/execute.c

```diff
@@ -5145,6 +5226,24 @@ static int exec_child(
                                 LOG_UNIT_MESSAGE(unit, "Executing: %s", line));
         }
 
+        /* If we are a user manager and setting up a UID/GID full mapping from a user namespace PID1 created for us,
+         * after we are done setting up we want to go back to the original UID/GID, given we'll be root when we
+         * move into the namespace. The point of having PID1 set us a mapping for us is to make sure there's no
+         * visible difference for the units due to the implicit user namespace. */
+        if (!MANAGER_IS_SYSTEM(unit->manager) && userns_set_up && !context->private_users) {
```

Comment:
instead of basically duplicating part of the check above, please set a new boolean var up top and check that here. i.e. `set_gid_uid_again = true` up top, or so.

these checks tend to be updated sooner or later, and thus if somebody accidentally updates one but not the other bad things will happen.